server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -87,7 +89,7 @@ public void openConnection(DiscoveryNode node, ConnectionProfile profile, Action
     @Override
     public Transport.Connection getConnection(DiscoveryNode node) {
         try {
-            return delegate.getConnection(node);
+            return getConnectionInternal(node);
         } catch (NodeNotConnectedException e) {
             return new ProxyConnection(getAnyRemoteConnection(), node);
```

Comment:
I think the functionality that you want is broken? We do not have a "direct" connection to the target node, the `ProxyConnection` type will be returned. But your `resolveRemoteClusterAlias` method does not unwrap proxy types.

Reply:
I thought I covered this at line 165 by calling unwrap before checking remote connection type: https://github.com/elastic/elasticsearch/blob/c4043826a12798080601076b98bf16b44e1231d3/server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java#L165

Is this okay or do you have something else in mind?

Reply:
There is also a test specifically for resolving the cluster alias of a proxy connections:

https://github.com/elastic/elasticsearch/blob/c4043826a12798080601076b98bf16b44e1231d3/server/src/test/java/org/elasticsearch/transport/RemoteConnectionManagerTests.java#L104-L107